src/DirectusAsset.tsx

```diff
@@ -1,11 +1,26 @@
+import * as React from 'react';
+
 import { DirectusAssetProps } from './types';
+import { DirectusContext } from './DirectusProvider';
 
-export const DirectusAsset = ({ apiUrl, asset, download = false, render }: DirectusAssetProps): JSX.Element => {
+export const DirectusAsset = ({
+  apiUrl: propsApiUrl,
+  asset,
+  download = false,
+  render,
+}: DirectusAssetProps): JSX.Element => {
+  const directusContext = React.useContext(DirectusContext);
   let params = {};
   if (download) {
     params = { ...params, download: '' };
   }
 
+  if (!directusContext && !propsApiUrl) {
+    throw new Error('DirectusAsset requires either a DirectusProvider or an apiUrl prop');
+  }
+
+  const apiUrl = propsApiUrl || directusContext?.apiUrl;
```

Comment:
Isn't typescript smart to infer directusContext? Is the optional chaining operator needed?

Reply:
Yes, for some reson it was giving me an error at the time i implemented it. But checking it again it's gone. Maybe Ts cached something there.